tests/tools/fswalker/fswalker_test.go

```diff
@@ -531,3 +546,58 @@ func Test_validateReport(t *testing.T) {
 		}
 	}
 }
+
+func Test_rerootWithCheckRename(t *testing.T) {
+
+	for _, tt := range []struct {
+		name     string
+		file     *fspb.File
+		newRoot  string
+		diffItem string
+		want     bool
+	}{
+		{
+			name: "diff item refers to root dir",
+			file: &fspb.File{
+				Path: "/some/absolute/path/to/source",
+				Info: &fspb.FileInfo{
+					Name:  "source",
+					IsDir: true,
+				},
+			},
+			newRoot:  "/some/absolute/path/to/source",
```

Comment:
Should this be `"/some/absolute/path/to/target"` ?

Reply:
No, sorry I think the `diffItem` text might have been confusing in this case. This is testing the conversion of paths to be relative to the root of the walk. So this entry `fspb.File` corresponds to the source directory, the root of the walk. So when we reroot every `fspb.File` we want it relative to this source. For a file with path `/some/absolute/path/to/source/subdir/file` we are looking for the relative path `"subdir/file"`, which we expect to cause `isRootDirectoryRename` to return `false`. And likewise for the source dir, we're looking for `"."`, which we expect it to return `true`.

Reply:
in fact the remainder of the text of `diffItem` is irrelevant as long as it starts with `"name: "`, since that is what triggers looking into the data structures for the rest of the filter check.

Reply:
I see what you mean. If you think of this as the restore's walk+report, it would make sense that the path you just walked is "target" (then we should change the `Path` and `Name` field above to reflect that). I was thinking more in terms of this `fspb.File` refers to the original walk, which gets rerooted, then fed (on a later restore) into `fswalker.ActionData` as the `Before` field (simulated in this test). Again though, the string paths themselves are arbitrary